lib/build_manager.py

```diff
@@ -33,20 +33,17 @@ def __init__(self, packages_names, distro):
 
     def __call__(self):
         try:
-            self.prepare_packages()
-
+            error_number = super(BuildManager, self).__call__()
         # distro related issues
         except (exception.DistributionNotSupportedError,
                 exception.DistributionVersionNotSupportedError,
                 exception.DistributionDetectionError) as exc:
             LOG.exception("Error during distribution detection. "
                           "See the logs for more information")
             return exc.errno
-        # package issues
-        except exception.PackageError as exc:
-            LOG.exception("Failed to load the package in components. "
-                          "See the logs for more information")
-            return exc.errno
+        else:
+            if error_number:
+                return error_number
 
         return self.build()
 
```

Comment:
ops sorry, the way github shows it in the conversation tab sucks for commenting in things that were not changed.
I was refering to `build` and `prepare_packages` (I think you can visualize them in this link https://github.com/open-power-host-os/builds/pull/105/commits/a3fc76ab4b8176c6688e94167fea7957f77c3a29#diff-d4f29a8a05d49edd09eeffae2d56abe5R54) 

> One more think that I think could be changed is the call method. It's kind of confusing to call BuildManager()() and calling a named method like "build" or "prepare_packages" is more straightfoward. That can be done later, if you agree.
+1